src/engine/strat_engine/dmnames.rs

```diff
@@ -88,12 +88,17 @@ impl Display for CacheRole {
 /// Format a name & uuid for the flex layer.
 /// Prerequisite: len(format!("{}", FORMAT_VERSION)) < 72
 pub fn format_flex_ids(pool_uuid: PoolUuid, role: FlexRole) -> (DmNameBuf, DmUuidBuf) {
-    let value = format!("stratis-{}-{}-flex-{}",
-                        FORMAT_VERSION,
-                        pool_uuid.simple().to_string(),
-                        role);
-    (DmNameBuf::new(value.clone()).expect("FORMAT_VERSION display length < 72"),
-     DmUuidBuf::new(value).expect("FORMAT_VERSION display length < 73"))
+
+    (DmNameBuf::new(format!("stratis-{}-{}-flex-{}",
+                            FORMAT_VERSION,
+                            pool_uuid.simple().to_string(),
+                            role))
+             .expect("FORMAT_VERSION display length < 72"),
+     DmUuidBuf::new(format!("stratis-{}-private-{}-flex-{}",
+                            FORMAT_VERSION,
+                            pool_uuid.simple().to_string(),
+                            role))
+             .expect("FORMAT_VERSION display length < 81"))
```

Comment:
Because you put more stuff into the UUID the FORMAT_VERSION for the UUID is constrained to be shorter than for the name. Length needs to be less than 63, not less than 81. Header should be updated appropriately, with the lower bound, i.e, 63 in this case. Others need to be changed as appropriate.